app/helpers/visit_helper.rb

```diff
@@ -18,44 +16,45 @@ def prison_name
   end
 
   def prison_names
-    Rails.configuration.prison_data.keys.sort
+    Prison.names
   end
 
   def prison_data(source=visit)
-    Rails.configuration.prison_data.fetch(source.prisoner.prison_name.to_s)
+    prison_name = source.is_a?(String) ? source : source.prisoner.prison_name.to_s
```

Comment:
I've changed prison_data in the initializer and Prison model.  I'll do it in the `visit_helper.rb` tomorrow before I rebase against master.